### ISSUE TITLE
Windows service sometimes fails to start mining on restart

### ISSUE THREAD

**Issue by plavirudar** [sp-hash/TeamBlackMiner#138]
I am using the program nssm to run the miner as a service. This works fine when the computer is started normally, but sometimes after a restart due to crash or power loss, the program does not start mining or retry connecting to the pool. It also does not log anything in the regular log file created with the `--log` option. 

This behavior is restricted to TBM, all other miners that I have used (gminer, t-rex, nbminer, TRM etc) do not have this problem and correctly restart as a service after a rig restart. 

I have attached the stdout information captured by nssm below. Note that the miner stops responding after the first failure to connect, and the only line that is printed is the shutting down threads line after the service is restarted: 

```
[1;36m00:00:00[0m [1m[2021-11-17 11:22:39.[0m[0m112[0m[1m][0m [0mCannot connect to mining pool. Retrying...[0m
[1;36m00:00:00[0m [1m[2021-11-17 11:22:39.[0m[0m112[0m[1m][0m [0mSleeping for 15 seconds[0m

[1;36m00:00:00[0m [1m[2021-11-17 11:22:39.[0m[0m128[0m[1m][0m [32mGPU0 Started CL work thread[0m
[1;36m00:00:00[0m [1m[2021-11-17 11:22:39.[0m[0m253[0m[1m][0m [32mGPU1 Started Cuda work thread[0m
[1;36m00:00:08[0m [1m[2021-11-17 11:31:28.[0m[0m499[0m[1m][0m [36mShutting down threads[0m
```

**Comment by sp-hash** [comment]
on disconnection the miner will retry for some times before it exits. You can change the number of retries with the

-r --retries                Set number of retry attempts for connecting to pool.

When launching the miner we reccomend to use a loop script like in the example bat files.

**Comment by plavirudar** [comment]
Thanks for the reply. 

The service functions like a loop script, but better, in that it automatically starts on boot and will restart the program when it exits for any reason. 

However the issue here is that the miner binary does not terminate and therefore the service gets stuck running, without either retrying or exiting. In the log above, you can see that at 11:22:39 it says sleeping, and then does nothing while the binary continues to execute and deadlock, the GPUs are not running and there is no connection to the pool. Only after I manually shut down the miner at 11:31:28 does it exit.

**Comment by sp-hash** [comment]
we discoverd a bug in 1.25. On bad internet connection sometimes the miner was sleeping for 3 minutes. fixed in v1.26
